diff -Naur gcc-6.3.0.old/gcc/ubsan.c gcc-6.3.0/gcc/ubsan.c
--- gcc-6.3.0.old/gcc/ubsan.c	2016-12-07 19:39:29.000000000 -0300
+++ gcc-6.3.0/gcc/ubsan.c	2017-07-23 02:55:57.764668196 -0300
@@ -1471,7 +1471,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
diff -Naur gcc-6.3.0.old/gcc/config/m68k/m68k.h gcc-6.3.0/gcc/config/m68k/m68k.h
--- gcc-6.3.0.old/gcc/config/m68k/m68k.h	2016-01-04 15:30:50.000000000 +0100
+++ gcc-6.3.0/gcc/config/m68k/m68k.h	2020-08-31 20:54:21.229514835 +0200
@@ -313,6 +313,8 @@
 
 #define INT_TYPE_SIZE (TARGET_SHORT ? 16 : 32)
 
+#define BOOL_TYPE_SIZE 16
+
 /* Define these to avoid dependence on meaning of `int'.  */
 #define WCHAR_TYPE "long int"
 #define WCHAR_TYPE_SIZE 32

